src/lib/transport.js

```diff
@@ -203,6 +204,15 @@ Transport.prototype.request = function (params, cb) {
     requestTimeout = params.requestTimeout;
   }
 
+  const pingRequest = params.path === '/' && params.method === 'HEAD';
+  if (pingRequest) {
+    const pingParam =
+      params.hasOwnProperty('pingTimeout') && params.pingTimeout;
+    const requestParam =
+      params.hasOwnProperty('requestTimeout') && params.requestTimeout;
+    requestTimeout = pingParam || requestParam || this.pingTimeout || requestTimeout;
```

Comment:
I believe the last `requestTimeout` should be `this.requestTimeout` which is set on line 39. Otherwise, the `requestParam` and the `requestTimeout` are the same value, additionally, we will never get to the `requestTimeout` since we default the `pingTimeout` to 3000.

What about ditching the requestTimeout from this altogether. Use whats passed in the params or default to the config?

```
const isPingRequest = params.path === '/' && params.method === 'HEAD'

if (isPingRequest) {
  requestTimeout = params.hasOwnProperty('pingTimeout') ? params.requestTimeout : this.pingTimeout;
}
```

Actually, how do I even pass in the pingTimeout in the params?

Reply:
yeah this gets a little confusing :).  requestTimeout is set  on line 152 to this.requestTimeout.  it's potentially overridden on line 203 by the requestTimeout param.  we don't want to use this.requestTimeout because it falls back to the global default of 30000ms or instantiated timeout, when we want to fallback to the global pingTimeout

technically i can remove it entirely,, because pingTimeout has a default now.  would that work?

pingParam is function param OR a default included in the generated spec
```
client.ping({ pingTimeout: 3000 })
```

this.pingTimeout is the global default of 3000ms, defined by the client  OR overridden on client instantiation.

so we get an ordering of
1) pingTimeout param on execute
1) requestTimeout param on execute
1) pingTimeout param on client instantiation
1) global default of 3000ms